hardware/esp8266com/esp8266/libraries/ESP8266WiFi/src/WiFiClient.cpp

```diff
@@ -105,11 +110,14 @@ int ICACHE_FLASH_ATTR  WiFiClient::connect(IPAddress ip, uint16_t port)
             g_localPort = MIN_LOCAL_PORT;
         if (err == ERR_OK)
             break;
-        if (err == ERR_USE)
+        if (err == ERR_USE) {
+        	esp_yield();
```

Comment:
Shouldn't we also call esp_schedule before calling esp_yield? Otherwise the sketch task will never be scheduled.